src/System.Security.Cryptography.Encoding/tests/Base64TransformsTests.cs

```diff
@@ -61,7 +60,7 @@ public void InvalidInput_ToBase64Transform()
                 InvalidInput_Base64Transform(transform);
 
                 // These exceptions only thrown in ToBase
-                AssertExtensions.Throws<ArgumentOutOfRangeException>("offsetOut", () => transform.TransformFinalBlock(data_5bytes, 0, 5));
+                AssertExtensions.Throws<ArgumentException>("inputCount", () => transform.TransformFinalBlock(data_5bytes, 0, 5));
```

Comment:
Is this a breaking change? According to the documentation of this class, it's OK to throw this exception.

The previous exception wasn't thrown from this code, but from an API this code called. This API isn't called anymore, hence the change.

Reply:
This test will now fail when run against .NET Framework (for compatibility tests) and needs to be fixed.

Reply:
Also, restore to ArgumentOutOfRangeException.

Reply:
> This test will now fail when run against .NET Framework

You mean as the parametername changed from `offsetOut` to `inputCount`?

If so, it feels strange to throw an exception indicating parameter `offsetOut` is out of range, when there is no parameter `offsetOut`.

Reply:
Until recently these tests also ran against .NET Framework to ensure that we were maintaining functional parity; so tests had to handle when .NET Framework and .NET Core differed on things like fixing the parameter name in the exception.

Fixing up the exposed parameter name in .NET Core is something that seems reasonable to me.  (The only runtime flow it would break is someone catching the argument exception and reacting to the parameter name... which is much more degenerate than someone who wrote `catch (ArgumentOutOfRangeException) {}` being broken because the exception type changed)

Reply:
(To clarify my wordy answer: The exception TYPE needs to be changed back, the parameter name change is good.  The part where the test needs to be adjusted to still work for .NET Framework is out of date, disregard.)